doc/manual/src/contributing/hacking.md

```diff
@@ -71,21 +104,29 @@ To install it in `$(pwd)/outputs` and test it:
 [nix-shell]$ make install
 [nix-shell]$ make installcheck -j $NIX_BUILD_CORES
 [nix-shell]$ ./outputs/out/bin/nix --version
-nix (Nix) 3.0
+nix (Nix) 2.12
 ```
 
-If you have a flakes-enabled Nix you can replace:
+To build Nix for the current operating system/architecture use
 
 ```console
-$ nix-shell
+$ nix-build
 ```
 
-by:
+This will build `packages.x86_64-linux.default` attribute defined in the `flake.nix`
+file. To build for other platforms add one of the following suffixes to
+it: aarch64-linux, i686-linux, x86\_64-darwin, x86\_64-linux. i.e.
 
 ```console
-$ nix develop
+$ nix-build -A packages.x86_64-linux.default
 ```
 
+## Developing
+`clangd` and `bear` are installed by default in the `devShell`. To use LSP with your editor, after running the above `configure` steps,
```

Comment:
```suggestion

`clangd` and `bear` are installed by default in the `devShell`. To use LSP with your editor, after running the above `configure` steps,
```

Please link to LSP, not everyone who wants to work on this will necessarily even know what that is.